src/main/scala/sigmastate/utils/SigmaByteReader.scala

```diff
@@ -62,8 +103,14 @@ class SigmaByteReader(val r: Reader,
     this
   }
 
-  @inline def getType(): SType = TypeSerializer.deserialize(this)
-  @inline def getValue(): SValue = ValueSerializer.deserialize(this)
+  @inline def getType(): SType = {
+    checkPositionLimit()
```

Comment:
this call is not required, since any first read inside TypeSerializer will be checked.

Reply:
Done.